notify/notify.go

```diff
@@ -242,21 +297,21 @@ func (n *DedupingNotifier) hasUpdate(alert *types.Alert, last *types.Notificatio
 	return true
 }
 
-// Notify implements the Notifier interface.
-func (n *DedupingNotifier) Notify(ctx context.Context, alerts ...*types.Alert) error {
+// ExtractNewNotifies filters out the notifications that shall be sent
```

Comment:
That rather describes the behavior this part had early on. Today it basically just checks whether even slightest thing changed in our alert batch and just resends everything if that's the case.